validation/coupled_models/coupled_atmosphere_ocean_model.jl

```diff
@@ -0,0 +1,75 @@
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Grids: topology
+
+#####
+##### Utilities
+#####
+
+using Oceananigans.Models: AbstractModel
+
+import Oceananigans.TimeSteppers: time_step!, update_state!
+import Oceananigans: fields
+
+struct CoupledAtmosphereOceanModel{O, A, C, P} <: AbstractModel{Nothing}
+    atmos :: A
+    ocean :: O
+    clock :: C
+    air_sea_flux_parameters :: P
+end
+
+function CoupledAtmosphereOceanModel(atmos, ocean; air_sea_flux_parameters=(; cᴰ=2e-3, ρ_atmos=1, ρ_ocean=1024))
+    clock = atmos.clock
+    return CoupledAtmosphereOceanModel(atmos, ocean, clock, air_sea_flux_parameters)
+end
+
+fields(model::CoupledAtmosphereOceanModel) = fields(model.atmos) # convenience hack for now
+
+function update_state!(coupled_model::CoupledAtmosphereOceanModel, update_atmos_ocean_state=true)
+    atmos_model = coupled_model.atmos
+    ocean_model = coupled_model.ocean
+
+    if update_atmos_ocean_state
+        update_state!(ocean_model)
+        update_state!(atmos_model)
+    end
+
+    uo, vo, wo = ocean_model.velocities
+    ua, va, wa = atmos_model.velocities
+    atmos_grid = atmos_model.grid
+    atmos_surface_flux_u = ua.boundary_conditions.bottom.condition
```

Comment:
Of course we don't need to have the same vertical grids for the atmosphere and oceans but for the moment do we need to have the same horizontal grid?  If so maybe in the future we would want to allow for different horizontal grids and average where needed?

Reply:
For sure, that would be an important feature if this moves beyond the prototype stage. I think folks are working on that problem (this prototype is only meant to inform their work --- we shouldn't implement coupled models directly in Oceananigans).